src/core/net/dnssd_server.cpp

```diff
@@ -1287,6 +1386,574 @@ void Server::ResetUpstreamQueryTransaction(UpstreamQueryTransaction &aTxn, Error
 }
 #endif
 
+#if OPENTHREAD_CONFIG_DNSSD_DISCOVERY_PROXY_ENABLE
+
+Server::DiscoveryProxy::DiscoveryProxy(Instance &aInstance)
+    : InstanceLocator(aInstance)
+    , mIsRunning(false)
+{
+}
+
+void Server::DiscoveryProxy::UpdateState(void)
+{
+    if (Get<Server>().IsRunning() && Get<Dnssd>().IsReady() && Get<BorderRouter::InfraIf>().IsRunning())
+    {
+        Start();
+    }
+    else
+    {
+        Stop();
+    }
+}
+
+void Server::DiscoveryProxy::Start(void)
+{
+    VerifyOrExit(!mIsRunning);
+    mIsRunning = true;
+    LogInfo("Started discovery proxy");
+
+exit:
+    return;
+}
+
+void Server::DiscoveryProxy::Stop(void)
+{
+    VerifyOrExit(mIsRunning);
+
+    for (ProxyQuery &query : Get<Server>().mProxyQueries)
+    {
+        Get<Server>().Finalize(query, Header::kResponseSuccess);
+    }
+
+    mIsRunning = false;
+    LogInfo("Stopped discovery proxy");
+
+exit:
+    return;
+}
+
+void Server::DiscoveryProxy::Resolve(ProxyQuery &aQuery, ProxyQueryInfo &aInfo)
+{
+    ProxyAction action;
```

Comment:
Should ```action``` be intialized with ```kNoAction``` or add ```default``` statement in switch case to set it to ```kNoAction```?  Compilers may warn that it may be used uninitialized.

Reply:
Hi @Cristib05 . Sorry. I somehow missed your comment above. 

> Compilers may warn that it may be used uninitialized.

Sounds reasonable. Will init it to be safe.

-----

Updating this PR is in my list of todos. Should be able to get to it (hopefully) soon. :)

Reply:
Thanks for your updates!